master/buildbot/process/buildstep.py

```diff
@@ -658,12 +658,25 @@ def start(self):
         raise NotImplementedError("your subclass must implement run()")
 
     def interrupt(self, reason):
+        # TODO: consider adding an INTERRUPTED or STOPPED status to use
```

Comment:
Sorry, I missed this comment.

I think that would be valuable, but I'd rather do it in a new pull-request because that is a different issue from fixing the interrupt logic.